### PR TITLE
[enterprise-metrics] Use memberlist instead of Consul for ring kv store

### PR DESCRIPTION
**Pull request by jdbaldry**
## 1.2.0

* [CHANGE] The chart now uses memberlist for the ring key-value store removing the need to run Consul. #340
  > **Warning:** Existing clusters will need to follow an upgrade procedure.
  > **Warning:** Existing clusters should first be upgraded to `v1.1.1` and use that version for migration before upgrading to `v1.2.0`.
  To upgrade to using memberlist:
  1. Ensure you are running the `v1.1.1` version of the chart.
  2. Deploy runtime `multi_kv_config` to use Consul as a primary and memberlist as the secondary key-value store.
     The values for such a change can be found in the [`multi-kv-consul-primary-values.yaml`](./multi-kv-consul-primary-values.yaml).
  3. Verify the configuration is in use by querying the [Configuration](https://cortexmetrics.io/docs/api/#configuration) HTTP API endpoint.
  4. Deploy runtime `multi_kv_config` to use memberlist as the primary and Consul as the secondary key-value store.
     The values for such a change can be found in [`multi-kv-memberlist-primary-values.yaml`](./multi-kv-memberlist-primary-values.yaml)
  5. Verify the configuration is in use by querying the [Configuration](https://cortexmetrics.io/docs/api/#configuration) HTTP API endpoint.
  6. Deploy `v1.2.0` helm chart which configures memberlist as the sole key-value store and removes the Consul resources.

### CODE REVIEW

File: charts/enterprise-metrics/CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+## 1.1.1
+
+* [FEATURE] Facilitate some runtime configuration of microservices. #342
+* [FEATURE] Upgrade to [Grafana Enterprise Metrics v1.2.0](https://grafana.com/docs/metrics-enterprise/latest/downloads/#v120----march-10-2021). #342
+
+## 1.1.0
+
+* [CHANGE] The memcached chart from the deprecated Helm stable repository has been removed and replaced with a Bitnami chart. #333
+  > **Warning:** This change will result in then cycling of your memcached Pods and will invalidate the existing cache.
```
Comment (MichelHollands) [inline charts/enterprise-metrics/CHANGELOG.md:38]:
```suggestion
  > **Warning:** This change will result in the cycling of your memcached Pods and will invalidate the existing cache.
```

File: charts/enterprise-metrics/CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+* [CHANGE] Memcached Pod resource limits have been lowered to match requests. #333
+* [FEATURE] YAML exports have been created for all chart values files. #333
+* [BUGFIX] The values for the querier/ruler/store-gateway `-<prefix>.memcached.max-item-size` have been corrected to match the limit configuredon the memcached server. #333
```
Comment (MichelHollands) [inline charts/enterprise-metrics/CHANGELOG.md:41]:
```suggestion
* [BUGFIX] The values for the querier/ruler/store-gateway `-<prefix>.memcached.max-item-size` have been corrected to match the limit configuration the memcached server. #333
```

File: charts/enterprise-metrics/CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+## Unreleased
+
+## 1.2.0
+
+* [CHANGE] The chart now uses memberlist for the ring key-value store removing the need to run Consul. #340
+  > **Warning:** Existing clusters will need to follow an upgrade procedure.
+  > **Warning:** Existing clusters should first be upgraded to `v1.1.1` and use that version for migration before upgrading to `v1.2.0`.
+  To upgrade to using memberlist:
+  1. Ensure you are running the `v1.1.1` version of the chart.
+  2. Deploy runtime `multi_kv_config` to use the Consul as a primary and memberlist as the secondary key-value store.
```
Comment (MichelHollands) [inline charts/enterprise-metrics/CHANGELOG.md:22]:
```suggestion
  2. Deploy runtime `multi_kv_config` to use Consul as a primary and memberlist as the secondary key-value store.
```

File: charts/enterprise-metrics/exports/default/apps-v1.Deployment-enterprise-metrics-admin-api.yaml [hunk 4]
```diff
@@ -0,0 +1,118 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    app: enterprise-metrics-admin-api
+    app.kubernetes.io/managed-by: Helmraiser
```
Comment (MichelHollands) [inline charts/enterprise-metrics/exports/default/apps-v1.Deployment-enterprise-metrics-admin-api.yaml:6]: Is Helmraiser a real tool?

**Review by MichelHollands — APPROVED** [review]

File: charts/enterprise-metrics/exports/default/apps-v1.Deployment-enterprise-metrics-admin-api.yaml (hunk 4 above)
Comment (jdbaldry) [reply charts/enterprise-metrics/exports/default/apps-v1.Deployment-enterprise-metrics-admin-api.yaml:6]: Yeah! It's a cool feature of Tanka to work with Helm charts (https://github.com/grafana/tanka/commit/6d99591bf404393da3e0904cc8e04e7fda42a354)

**Review by MichelHollands — APPROVED** [review]
LGTM

**Pull request merged**